packages/adapter-utils/src/element_source.ts

```diff
@@ -67,10 +67,14 @@ export const buildElementsSourceFromElements = (
     }
   }
 
-  const get = async (id: ElemID): Promise<Value> => {
+  const getElement = (id: ElemID): Value => {
     const { parent } = id.createTopLevelParentID()
     const topLevelElement = elementsMap[parent.getFullName()]
-    const element = elementsMap[id.getFullName()] ?? (topLevelElement && resolvePath(topLevelElement, id))
+    return elementsMap[id.getFullName()] ?? (topLevelElement && resolvePath(topLevelElement, id))
```

Comment:
`resolvePath` can handle the ID being the top level ID, no need to perform another lookup in the elements map
```suggestion
    return topLevelElement && resolvePath(topLevelElement, id)
```

Reply:
Theoretically, `elements` may contain fields without their parent types.

Reply:
really?
if that is the case, then the code should have taken `baseID`

Reply:
AFAICT all places in the code only pass in top level elements.

maybe worth adding a warn log if someone tries to build an element source here with something that isn't a top level element?

Reply:
I prefer checking them both, as I see a situation when somebody uses this function in deploy, where changes have fields in them.
```typescript
    const baseId = id.createBaseID().parent
    const topLevelParentId = id.createTopLevelParentID().parent
    const element = elementsMap[baseId.getFullName()] ?? elementsMap[topLevelParentId.getFullName()]
    return element && resolvePath(element, id)
```

Reply:
I agree with @dantal4